Mastodon/Protocol/Provider/DataSourceFacade+Profile.swift

```diff
@@ -46,54 +46,83 @@ extension DataSourceFacade {
             account: redirectRecord
         )
     }
-    
+
     @MainActor
     static func coordinateToProfileScene(
         provider: ViewControllerWithDependencies & AuthContextProvider,
-        user: ManagedObjectRecord<MastodonUser>
+        username: String,
+        domain: String
     ) async {
-        guard let user = user.object(in: provider.context.managedObjectContext) else {
-            assertionFailure()
-            return
+        provider.coordinator.showLoading()
+
+        Task {
+            do {
+                guard let account = try await provider.context.apiService.fetchUser(username: username,
+                                                                                    domain: domain,
+                                                                                    authenticationBox: provider.authContext.mastodonAuthenticationBox) else {
+                    return provider.coordinator.hideLoading()
+                }
```

Comment:
I know this Is probably a controversial thing, but I really think the following is much more readable (less indentations )

```suggestion
                guard let account = try await provider.context.apiService.fetchUser(
                    username: username,
                    domain: domain,
                    authenticationBox: provider.authContext.mastodonAuthenticationBox
                ) else {
                    return provider.coordinator.hideLoading()
                }
```